napari/layers/base/base.py

```diff
@@ -547,7 +547,7 @@ def thumbnail(self, thumbnail):
     @property
     def ndim(self):
         """int: Number of dimensions in the data."""
-        return self.dims.ndim
+        return self._dims.ndim
 
```

Comment:
```suggestion

@property
def dims()
        warnings.warn(
            "dims parameter is deprecated and will be removed in version 0.4.1",
            DeprecationWarning
        )
        return self._dims

```

This suggestion does not contain needed imports.

Reply:
Ooo, I like this idea. Thanks for the suggestion @Czaki. Thoughts @jni? Maybe we get this in before `0.4.0`

Reply:
This would be great, thanks @Czaki for the suggestion.

Reply:
done